#OICI-2  #OICI-15 Implementation of Comores features

kevel-dev opened the pull request.

mngoe committed:
Merge pull request #6 from openimis/main

Merge openIMIS Main to Fork main

kevel-dev committed:
Merge pull request #19 from openimis/main

Main

kevel-dev committed:
Merge pull request #21 from openimis/main

Main

delcroip committed:
Merge pull request #111 from openimis/develop

MERGING develop into release/23.10

hirensoni913 committed:
Merge pull request #112 from openimis/release/23.10

MERGING release/23.10 into main

delcroip committed:
Merge pull request #125 from openimis/develop

fix family

mngoe committed:
Merge pull request #32 from openimis/develop

Merge des develop

Keveltiah committed:
Ajout champ niveau et parent a merger ref28214

Keveltiah committed:
Ajout champs specifique comores

mngoe committed:
Merge pull request #33 from mngoe/feature-28214

Ajout champ niveau et parent a merger ref28214

mngoe committed:
Merge pull request #34 from mngoe/feature-28214-comores

Feature 28214 comores

Keveltiah committed:
Ajout des niveaux de revenus ref28215

Keveltiah committed:
Ajout queries

mngoe committed:
Merge pull request #35 from mngoe/feature-28215

Ajout des niveaux de revenus ref28215

Keveltiah committed:
Ajout autres champs specifiques

Keveltiah committed:
pull develop*cores

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into feature-28216

Keveltiah committed:
feat: generation du chfid ref28216

mngoe committed:
Merge pull request #37 from mngoe/feature-28216

feat: generation du chfid ref28216

Keveltiah committed:
fix: Génération du chfid uniquement si c'est la conf de comores ref28216

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into feature-28214-comores

Keveltiah committed:
feat: Ajout champ incomelevel sur assuré ref28214

Keveltiah committed:
feat: gestion suppression et ajout d'une famille sur une famille parente ref28090

Keveltiah committed:
fix: Ajustement documentation

mngoe committed:
Merge pull request #39 from mngoe/feature-28090

feat: gestion suppression et ajout d'une famille sur une famille pare…

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into feature-28214-comores

Keveltiah committed:
Feat Ajout filtre sur parent reffeature-28214

delcroip committed:
Merge pull request #126 from openimis/develop

MERGING develop into release/24.04

mngoe committed:
Merge pull request #38 from mngoe/feature-28216

fix: Génération du chfid uniquement si c'est la conf de comores ref28216

mngoe committed:
Merge pull request #36 from mngoe/feature-28214-comores

Ajout autres champs comores

delcroip committed:
Merge pull request #129 from openimis/develop

MERGING develop into release/24.04

Keveltiah committed:
feat: remove print

Keveltiah committed:
feat: remove print

mngoe committed:
Merge pull request #41 from mngoe/feature-28216

feat: remove print

kevel-dev committed:
Merge pull request #42 from openimis/main

feat: Merge Main into mngoe

mngoe committed:
Merge pull request #43 from openimis/release/24.04

Merge Release/24.04 into develop-comores

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into develop-comores

Keveltiah committed:
fix(Comores): Vérification si le chfid n'est pas renseigné en meme temps que la conf de comores n'est pas active ref28922

mngoe committed:
Merge pull request #47 from mngoe/feature-28922

fix(Comores): Vérification si le chfid n'est pas renseigné en meme te…

Keveltiah committed:
fix: Ajustement generation code

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into develop-comores

Keveltiah committed:
fix: Ajustement migrations

delcroip committed:
Merge pull request #135 from openimis/release/24.04

MERGING release/24.04 into main

kevel-dev committed:
fix: Ajustement migration

kevel-dev committed:
Merge branch 'develop-comores' into fix-migration

mngoe committed:
Merge pull request #51 from mngoe/fix-migration

Fix migration

kevel-dev committed:
fix: Ajustement migration

mngoe committed:
Merge pull request #52 from mngoe/fix-migration

fix: Ajustement migration

kevel-dev committed:
fix: Ajustement migration

mngoe committed:
Merge pull request #53 from mngoe/fix-migration

fix: Ajustement migration

mngoe committed:
Merge pull request #48 from mngoe/feature-28922

fix: Ajustement generation code

Le-positif-clinton committed:
feat: Requetes SQL pour inserer les valeurs et configurations prevues en database

Le-positif-clinton committed:
feat:ajout des revenus a la migration

Le-positif-clinton committed:
feat: Ajout Requetes SQL pour inserer les valeurs par defaut des families

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into develop-comores

Le-positif-clinton committed:
feat: mise a jour de tblFamilyTypes

mngoe committed:
Merge pull request #55 from mngoe/feature-29227

feat: Requetes SQL pour inserer les valeurs et configurations prevues…

kevel-dev committed:
Merge pull request #57 from openimis/main

Main

Keveltiah committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into develop-comores

Keveltiah committed:
debug

mngoe committed:
Merge pull request #65 from mngoe/debug

Debug

Keveltiah committed:
debug

Keveltiah committed:
debug

Keveltiah committed:
debug

Keveltiah committed:
debug

Keveltiah committed:
fix: Ajustement generation chfid

Keveltiah committed:
fix: Ajustement generation chfid

Keveltiah committed:
fix: Ajustement generation chfid

dragos-dobre commented on insuree/services.py:
```diff
@@ -310,8 +312,26 @@ def create_or_update(self, data):
             self.activate_policies_of_insuree(insuree, audit_user_id=data['audit_user_id'])
+        if "uuid" not in data:
+            if InsureeConfig.comores_features_enabled:
+                print("Auto genate CHFID")
```
Remove debug prints.

dragos-dobre commented on insuree/services.py:
```diff
@@ -339,6 +359,10 @@ def activate_policies_of_insuree(self, insuree, audit_user_id):
+        print("_create_or_update****")
+        if not InsureeConfig.comores_features_enabled:
+            if not insuree.chf_id:
+                raise Exception(f"Aucun CHFID renseigné pourtant le système n'a pas la configuration de Comores active")
```
Exception in English normally and use translations in Lokalise.

dragos-dobre commented on insuree/apps.py:
```diff
@@ -75,6 +76,7 @@ class InsureeConfig(AppConfig):
     insuree_fsp_mandatory = None
     insuree_as_worker = None
     is_insuree_photo_required = None
+    comores_features_enabled = None
```
This should be functionality specific and not implementation specific. It will be easier to integrate in the Master Version.

dragos-dobre commented on insuree/income_levels.sql:
```diff
@@ -0,0 +1,11 @@
+INSERT INTO tblIncomeLevels (FrenchVersion, EnglishVersion) VALUES
+('Néant', 'Nothing'),
```
This is managed now in the translation files. Please check other values like the professions or family types and  in Lokalise.

dragos-dobre commented on insuree/gql_mutations.py:
```diff
@@ -57,6 +57,12 @@ class InsureeBase:
     status = graphene.String(required=False)
     status_reason = graphene.String(required=False)
     status_date = graphene.Date(required=False)
+    poligamous = graphene.Boolean(required=False)
```
There are several attributes repeating in Family and Insuree, and I think this should not happen: poligamous, coordinates, preferred_payment_method, income_level_id, professional_situation, bank_coordinates.

Please fix spelling meeting polygamous.

dragos-dobre commented on insuree/gql_mutations.py:
```diff
@@ -324,6 +336,99 @@ def async_mutate(cls, user, **data):
+                errors.append({
+                    'title': child_family_uuid,
+                    'list': [{'message': _(
+                        "Impossible d'assigner la famille %(id)s à elle-même") % {'id': child_family_uuid}}]
```
Make the error message in EN and add the translation in Lokalise. Do the same for the other errors.

dragos-dobre commented on insuree/gql_queries.py:
```diff
@@ -39,6 +40,17 @@ class Meta:
+        interfaces = (graphene.relay.Node,)
+        filter_fields = {
+            "id":["exact"],
+            "french_version":["exact", "istartswith", "icontains", "iexact"],
```
You should only match based on id and not the value. Check also the other comment on the linked migration file.

dragos-dobre commented on insuree/models.py:
```diff
@@ -70,6 +70,13 @@ class Meta:
 
+class IncomeLevels(models.Model):
+    id  = models.AutoField(db_column='IncomeLevelID', primary_key=True)
+    french_version = models.CharField(db_column='FrenchVersion', max_length=200, blank=True, null=True)
```
If we want to integrate with MV then the fields names have to change as countries might use other languages. Maybe to use first_language and second_language. If you change the translation management to Lokalise, then you might need  only one language here to know what the id corresponds to.

dragos-dobre commented on insuree/migrations/0025_auto_20240731_1335.py:
```diff
@@ -0,0 +1,62 @@
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
To make is specific to an implementation, this should be in another Comoros specific module like openimis-be-insuree_comoros_py.

dragos-dobre commented on insuree/models.py:
```diff
@@ -283,6 +307,18 @@ def is_adult(self, reference_date=None):
                                       blank=True, null=True, related_name='insurees')
     audit_user_id = models.IntegerField(db_column='AuditUserID')
     # row_id = models.BinaryField(db_column='RowID', blank=True, null=True)
+    poligamous = models.BooleanField(
```
This should be only on Family level.

Le-positif-clinton committed:
Feat: Correction sur le PR comores insuree

Le-positif-clinton committed:
feat: custom_chif_id  and BaseInsureeFamily

kevel-dev committed:
Merge pull request #67 from mngoe/feature-29884

Feat: Correction sur le PR comores insuree

Le-positif-clinton committed:
feat: erreur config.no_chfid

Le-positif-clinton committed:
feat:iltre 'isSubfamily' dans la requete graphql qui récupère la liste des familles et resolution du problème du filtre familyType

kevel-dev committed:
Merge pull request #70 from mngoe/feature--30066

feat:iltre 'isSubfamily' dans la requete graphql qui récupère la list…

kevel-dev committed:
fix: Ajustement du bugn sur la photot de l'assuré ref30849

Le-positif-clinton committed:
feat: mise a jour du chef de famille sans modifier son chf_id

mngoe committed:
Merge pull request #92 from mngoe/feature-31051

feat: mise a jour du chef de famille sans modifier son chf_id

delcroip commented on insuree/gql_mutations.py:
```diff
@@ -57,6 +57,11 @@ class InsureeBase:
     status_date = graphene.Date(required=False)
+    coordinates = graphene.String(max_length=255, required=False)
+    preferred_payment_method = graphene.String(max_length=50, required=False)
+    income_level_id = graphene.Int(required=False)
```
The income level is not stable in time therefore should not be here

delcroip commented on insuree/gql_mutations.py:
```diff
@@ -57,6 +57,11 @@ class InsureeBase:
     status = graphene.String(required=False)
     status_reason = graphene.String(required=False)
     status_date = graphene.Date(required=False)
+    coordinates = graphene.String(max_length=255, required=False)
```
what is the difference with geolocation

delcroip commented on insuree/gql_mutations.py:
```diff
@@ -57,6 +57,11 @@ class InsureeBase:
     status_reason = graphene.String(required=False)
     status_date = graphene.Date(required=False)
+    coordinates = graphene.String(max_length=255, required=False)
+    preferred_payment_method = graphene.String(max_length=50, required=False)
```
Why String ? the system is likely to support only few methods,

delcroip commented on insuree/gql_mutations.py:
```diff
@@ -57,6 +57,11 @@ class InsureeBase:
+    coordinates = graphene.String(max_length=255, required=False)
+    preferred_payment_method = graphene.String(max_length=50, required=False)
+    income_level_id = graphene.Int(required=False)
+    professional_situation = graphene.String(max_length=255, required=False)
```
difference with profession_id ?

delcroip commented on insuree/gql_mutations.py:
```diff
@@ -83,9 +88,15 @@ class FamilyBase:
+    family_level = graphene.String(max_length=1, required=False)
+    parent_id = graphene.Int(required=False)
+    polygamous = graphene.Boolean(required=False)
+    coordinates = graphene.String(max_length=255, required=False)
+    preferred_payment_method = graphene.String(max_length=50, required=False)
+    income_level_id = graphene.Int(required=False)
+    professional_situation = graphene.String(max_length=255, required=False)
+    bank_coordinates = graphene.String(max_length=255, required=False)
```
Why having that here, why not taking head data ?

Le-positif-clinton committed:
feat: Ajout de la gestion des pièces jointes sur les ménages

Le-positif-clinton committed:
feat: adaptation de la génération du code d’immatriculation pour bénéficiaire sans NIN

mngoe committed:
Merge pull request #97 from mngoe/feature-31171

feat: Ajout de la gestion des pièces jointes sur les ménages

Le-positif-clinton committed:
fix: éviter l'exception en cas d'UUID manquant lors de la creation/mise a jour d'une famille

mngoe committed:
Merge pull request #99 from mngoe/feature-31172

feat: adaptation de la génération du code d’immatriculation pour bénéficiaire sans NIN

Le-positif-clinton committed:
fix: mise a jour pièce jointe (ajout sur Family au lieu de Insuree)

kevel-dev committed:
Merge pull request #100 from mngoe/feature-31171-fix

fix: mise a jour pièce jointe (ajout sur Family au lieu de Insuree)

Le-positif-clinton committed:
Feat: test unitaire mise a jour d'un assuré sans modification du chfid

mngoe committed:
Merge pull request #101 from mngoe/feature-31364

Feat: test unitaire mise a jour d'un assuré sans modification du chfid

Le-positif-clinton committed:
feat:  add GraphQL unit test for family attachments

mngoe committed:
Merge pull request #102 from mngoe/feature-31191

feat:  add GraphQL unit test for family attachments

kevel-dev committed:
feat(migration):rajout nouvelles de realtions, prefession etc

kevel-dev committed:
Merge pull request #104 from mngoe/feature-31956

feat(migration):rajout nouvelles de realtions, prefession etc

kevel-dev committed:
fix: Ajustement check date expiration avant traitement

kevel-dev committed:
feat:debug

Le-positif-clinton committed:
fix: modification des informations de sms d'une famille

kevel-dev committed:
Merge pull request #105 from mngoe/feature-32167

fix: modification des informations de sms d'une famille

kevel-dev committed:
Merge pull request #106 from mngoe/feature-32167

fix: modification des informations de sms d'une famille

Le-positif-clinton committed:
Fix test unitaires

kevel-dev committed:
Merge pull request #108 from mngoe/feature-32200

Fix test unitaires insuree

mngoe committed:
Merge pull request #107 from mngoe/feature-32167

fix: modification des informations de sms d'une famille

Le-positif-clinton committed:
feat: Correction sur le NIN backend

kevel-dev committed:
Merge pull request #109 from mngoe/feature-32299

feat: Correction sur le NIN backend

kevel-dev committed:
Merge pull request #110 from mngoe/feature-32299

feat: Correction sur le NIN backend

mngoe committed:
Merge pull request #111 from mngoe/feature-32299

feat: Correction sur le NIN backend

Le-positif-clinton committed:
feat: Améliorer la sélection de l'agent d'inscription lors de l'ajout de la photo de l'assuré

kevel-dev committed:
Merge pull request #112 from mngoe/feature-32306

feat: Améliorer la sélection de l'agent d'inscription lors de l'ajout…

Le-positif-clinton committed:
feat: Improve Enrolment Officer Selection When Adding Insuree Photo

mngoe committed:
Merge pull request #115 from mngoe/feature-32306

feat: Improve Enrolment Officer Selection When Adding Insuree Photo  #OP-2527

kevel-dev committed:
Merge pull request #113 from mngoe/feature-32306

Improve Enrolment Officer Selection When Adding Insuree Photo  #OP-2527

ArkeupIDComores committed:
Merge branch 'develop-comores' of https://github.com/mngoe/openimis-be-insuree_py into develop-comores

Le-positif-clinton committed:
fix: filtre de l'agent d'inscription par village de l'assuré

Iandraina-etech committed:
ajout des champs

ajout des champs:
- types_habitation
- maladie_invalidante_non
-couverture_assurance_mutuelle
-milieu_de_residence
-handicap_non

kevel-dev committed:
Merge pull request #117 from mngoe/feature-32306

fix: filtre de l'agent d'inscription par village de l'assuré

ArkeupIDComores committed:
Merge pull request #1 from ArkeupIDComores/feature-arkeup

adding new fields

kevel-dev committed:
fix: fix error insureePolicy returned more than one ref32447

kevel-dev committed:
Merge pull request #120 from mngoe/feature-32447

fix: fix error insureePolicy returned more than one ref32447

kevel-dev committed:
fix: fix error insureePolicy returned more than one ref32447

kevel-dev committed:
Merge pull request #121 from mngoe/feature-32447-test

fix: fix error insureePolicy returned more than one ref32447

kevel-dev committed:
fix: fix error insureePolicy returned more than one ref32447

mngoe committed:
Merge pull request #122 from mngoe/feature-32447-main

fix: fix error insureePolicy returned more than one ref32447

Iandraina-etech committed:
Changement en anglais des champs ajoutes et remove la fonction mapfieds et la classe stringOrint

traduction des champs en anglais  et amelioration du respect des normes CamelCase et snake_case
-Maladieinvalidante_Non=> NonDisablingDisease
- Handicap_Non=>NoDisability
-CouvertureAssuranceMutuelle=>MutualInsuranceCoverage
-TypesHabitation=>HousingType
-Milieuderesidence=>ResidenceEnvironment

adaptation des changement sur les types gql base et resolver

retrait de la fonction
def _map_fields(self, data): dans services jugé  non necessaire il suffit de suiver les noms des champs coté front end

retrait de class StringOrInt :qqlmutation il suffit de s assurer que un entier soit transmis sur le json

Iandraina-etech committed:
Update income_levels.sql

ArkeupIDComores committed:
Merge pull request #2 from ArkeupIDComores/adding new fields

adding new fields

ArkeupIDComores committed:
Merge branch 'develop-comores' into feature-arkeup

ArkeupIDComores committed:
Merge branch 'feature-arkeup' of https://github.com/ArkeupIDComores/openimis-be-insuree_py into feature-arkeup

ArkeupIDComores committed:
change ID for the field

mngoe committed:
Merge pull request #118 from mngoe/feature-32306

fix: Enrolment officer filter by insured village

Iandraina-etech committed:
Fix remarque PR maxime

fix remarques PR maxime
Feature arkeup : adding new fields
#123

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

Iandraina-etech committed:
Update schema.py

mngoe committed:
Merge pull request #123 from ArkeupIDComores/feature-arkeup

Feature arkeup : adding new fields

Le-positif-clinton committed:
Feat: Unit test for Enrolment Officer Selection When Adding Insuree Photo

kevel-dev committed:
Merge pull request #124 from mngoe/feature-32306

Feat: Unit test for Enrolment Officer Selection When Adding Insuree Photo

Le-positif-clinton committed:
Update test_graphql.py: remove  cls.test_village.parent.parent.code

Le-positif-clinton committed:
Update test_graphql.py

Le-positif-clinton committed:
fix: unit test test_insuree_officers_query

mngoe committed:
Merge pull request #125 from mngoe/feature-32306

Feat: Unit test for Enrolment Officer Selection When Adding Insuree Photo

mngoe committed:
Merge pull request #126 from mngoe/develop-comores

Develop comores to Test Comores

mngoe committed:
Merge pull request #129 from mngoe/test-comores

Test comores vers Live Comores

Iandraina-etech committed:
Adding required_fields to make the lot 15b work

kevel-dev committed:
Merge pull request #132 from Iandraina-etech/develop-comores

lot 15:calcul Automatique des cotisation

Iandraina-etech committed:
Add scoring migrations and update ScoreContributionMapping fields

Introduces several data migrations to populate scores for HousingType, IncomeLevels, and FamilySizeScores, and to map score ranges to ContributionPlans. Also changes the lower_born and higher_born fields in ScoreContributionMapping from IntegerField to DecimalField for more precise score range handling.

Iandraina-etech committed:
Update 0037_score_mapping.py

kevel-dev committed:
Merge pull request #133 from Iandraina-etech/develop-comores

lot 15: Data insertion =>Add scoring migrations and update ScoreContributionMapping fields

Iandraina-etech committed:
Add FamilyIncomeScores model and fix_income field

Introduces the FamilyIncomeScores model to store income-based scoring, adds the fix_income DecimalField to the Insuree model, and updates GraphQL mutations to support the new field. Includes migrations to create the new table, add the field, and populate initial income score mappings.

kevel-dev committed:
Merge pull request #134 from Iandraina-etech/develop-comores

Lot 15: Calcul automatique des contributions

Iandraina-etech committed:
Merge pull request #1 from Iandraina-etech/develop-comores

Develop comores

kevel-dev committed:
Merge pull request #135 from Iandraina-etech/test-comores

lot 15:Calcul Automatique des contributions

Iandraina-etech committed:
Merge pull request #2 from Iandraina-etech/test-comores

lot 15 :features

mngoe committed:
Merge pull request #136 from Iandraina-etech/main-comores

lot 15 :calcul auto contribution

kevel-dev committed:
feat(AffiliationType): add filter for affiliation types ref37415

kevel-dev committed:
Merge pull request #137 from mngoe/feature-37415

feat(AffiliationType): add filter for affiliation types ref37415

kevel-dev committed:
feat: ajustement filters on familly type ref37415

kevel-dev committed:
Merge pull request #140 from mngoe/feature-37415

feat: ajustement filters on familly type ref37415
